hicommonwealth/commonwealth
# RESTful Community Routes

## Link to Issue
Closes: #4805

## Description of Changes
- Updates the `chain` routes to become `community` routes with the following schema:
  - Create community: `POST /communities`
  - Update community: `PATCH /communities/:id`
  - Get communities `GET /communities?snapshots=true&search=XXX`
  - Delete community: `DELETE /communities/:id`
  - Create node: `POST /nodes`
  - Get nodes: `GET /nodes`
  - Get community analytics: `GET /communities/:id/stats`

## Test Plan
- Existing API/E2E tests are update to reference new routes
- CA test locally
  - Load site– confirm that page loads without errors
  - Create a new community– confirm that it's created correctly
  - Update the new community– confirm that it's updated correctly
  - Search for the new community– confirm that it shows in the "communities" tab of the search page
  - Delete the new community– confirm that it's deleted
  - Create a new node via admin panel– confirm that it's created
  - On the admin page– confirm it shows per-community analytics correctly


## Deployment Plan
N/A

## Other Considerations
N/A
